.github/workflows/update-lockfiles.yml

```diff
@@ -0,0 +1,75 @@
+name: Update lockfiles
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: |
+          Branch or ref to clone. This is where the lockfiles will be added and
+          ultimately committed.
+        required: true
+
+      dry-run:
+        type: choice
+        options:
+          - "yes"
+          - "no"
+        default: "yes"
+        description: |
+          Dry run mode when enabled with the "yes" option will not commit or push
+          the generated files.
+
+
+jobs:
+
+  update-lockfiles:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.11"]
+        os: [macOS-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.ref }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install -r requirements-test.txt
+      # Windows and posix have different syntax for setting the outputs and env vars
+      # in github actions so we need to duplicate the next step one for windows and
+      # one for macOS.
+      - name: Regenerate windows files
+        if: runner.os == 'Windows'
+        run: |
+          python scripts/regenerate-lock-files --show-files
+          echo "CHANGES=$(git status --porcelain=v1)" >> $env:GITHUB_ENV
+          echo "CHANGES=$(git status --porcelain=v1)" >> $env:GITHUB_OUTPUT
+      - name: Regenerate lock files
+        if: runner.os != 'Windows'
+        run: |
+          python scripts/regenerate-lock-files --show-files
+          echo "CHANGES<<EOF" >> $GITHUB_ENV
```

Comment:
It might be helpful to have a comment on why we need the EOF lines here. It's hard to follow otherwise.